lib/roken/version-script.mapin

```diff
@@ -61,38 +56,74 @@ HEIMDAL_ROKEN_2.0 {
 		rk_estrdup;
 		rk_evasprintf;
 		rk_ewrite;
+#ifndef HAVE_FLOCK
 		rk_flock;
+#endif
+#ifndef HAVE_FNMATCH
 		rk_fnmatch;
+#endif
 		rk_free_environment;
+#ifndef HAVE_FREEADDRINFO
 		rk_freeaddrinfo;
+#endif
+#ifndef HAVE_FREEHOSTENT
 		rk_freehostent;
-		rk_freeifaddrs;
-		rk_gai_strerror;
+#endif
+#ifndef HAVE_GETADDRINFO
 		rk_getaddrinfo;
+#endif
 		rk_getauxv;
 		rk_getauxval;
+#ifndef HAVE_GETIFADDRS
 		rk_getifaddrs;
+#endif
+#ifndef HAVE_GETIPNODEBYADDR
 		rk_getipnodebyaddr;
+#endif
+#ifndef HAVE_GETIPNODEBYNAME
 		rk_getipnodebyname;
+#endif
+#ifndef HAVE_GETNAMEINFO
 		rk_getnameinfo;
+#endif
+#ifndef HAVE_GETPROGNAME
 		rk_getprogname;
-		rk_glob;
-		rk_globfree;
+#endif
 		rk_hex_decode;
 		rk_hex_encode;
 		rk_hostent_find_fqdn;
+#ifndef HAVE_INET_NTOP
 		rk_inet_ntop;
+#endif
+#ifndef HAVE_INET_PTON
 		rk_inet_pton;
+#endif
 		rk_injectauxv;
+#ifndef HAVE_LOCALTIME_R
 		rk_localtime_r;
-		rk_memmem;
```

Comment:
We do use `memmem()`/`rk_memmem()`.

Reply:
I deleted this because `memmem` is redirected to `rk_smemmem`. Take a look at https://github.com/heimdal/heimdal/blob/cb9a13032285ee9a20507f21c44b67a547a9996c/lib/roken/roken.h.in#L505.

Is this intentional? I don't find any other references to `rk_memmem` directly.

Reply:
Yes, it gets renamed to `rk_memmem()`, so the three uses in `kdc/connect.c` are actually uses of `rk_memmem()` (but only when the OS/C library doesn't have a `memmem()`).  That's intentional.

Reply:
Sorry for the annoyance and my possible misunderstanding of your comment, but could you point me to the definition of `rk_memmem`? Here's my understanding:

* Renaming of `memmmem` to `rk_smemmem` happens here: https://github.com/heimdal/heimdal/blob/cb9a13032285ee9a20507f21c44b67a547a9996c/lib/roken/roken.h.in#L506-L507

* Definition of `memmem` (which gets renamed again to `rk_smemmem` by the previous macro) is here: https://github.com/heimdal/heimdal/blob/cb9a13032285ee9a20507f21c44b67a547a9996c/lib/roken/memmem.c#L38-L41

* All calls of `memmem` happen in `kdc/connect.c`

Where does the renaming happen for `memmem` to `rk_memmem`? Note the difference of the letter `s` between `rk_memmem` and `rk_smemmem`.

Reply:
Argh, where did that `s` come from?  Well, that's clearly a typo.  Now I'm curious how this works on Windows.